yb-voyager/test/utils/cmdutils.go

```diff
@@ -0,0 +1,128 @@
+package testutils
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	testcontainers "github.com/yugabyte/yb-voyager/yb-voyager/test/containers"
+)
+
+func CreateTempExportDir() string {
+	// Create a temporary directory for export inside /tmp
+	exportDir, err := os.MkdirTemp("", "yb-voyager-export")
+	if err != nil {
+		utils.ErrExit("failed to create temp export dir for testing: %v", err)
+	}
+
+	return exportDir
+}
+
+func RemoveTempExportDir(exportDir string) {
+	// Remove the temporary directory
+	err := os.RemoveAll(exportDir)
+	if err != nil {
+		utils.ErrExit("failed to remove temp export dir: %v", err)
+	}
+}
+
+func RunVoyagerCommmand(container testcontainers.TestContainer,
+	cmdName string, cmdArgs []string,
+	doDuringCmd func(),
+) error {
+	// Gather DB connection info
+	host, port, err := container.GetHostPort()
+	if err != nil {
+		return fmt.Errorf("failed to get host port for container: %v", err)
+	}
+
+	config := container.GetConfig()
+
+	var connectionArgs []string
+	if isSourceCmd(cmdName) {
+		connectionArgs = []string{
+			"--source-db-type", config.DBType,
+			"--source-db-user", config.User,
+			"--source-db-password", config.Password,
+			"--source-db-schema", config.Schema,
+			"--source-db-name", config.DBName,
+			"--source-db-host", host,
+			"--source-db-port", strconv.Itoa(port),
+			"--source-ssl-mode", "disable",
+		}
+	} else if isTargetCmd(cmdName) {
+		connectionArgs = []string{
+			"--target-db-user", config.User,
+			"--target-db-password", config.Password,
+			"--target-db-schema", config.Schema,
+			"--target-db-name", config.DBName,
+			"--target-db-host", host,
+			"--target-db-port", strconv.Itoa(port),
+			"--target-ssl-mode", "disable",
+		}
+	}
+
+	// 1) Build the command to run.
+	cmdArgs = append(connectionArgs, cmdArgs...)
+	cmdStr := fmt.Sprintf("yb-voyager %s %s", cmdName, strings.Join(cmdArgs, " "))
+	cmd := exec.Command("/bin/bash", "-c", cmdStr)
```

Comment:
let's set SEND_DIAGNOSTICS env to false in the cmd here?

Reply:
No, no. We have set `YB_VOYAGER_SEND_DIAGNOSTICS=false` in the run-test scripts (so wherever they are run, they will not send diagnostics).

Reply:
oh wait, yes this test must have run a few times, worth checking!